packages/ember-application/tests/system/instance_initializers_test.js

```diff
@@ -1,410 +1,386 @@
 import { run } from 'ember-metal';
-import Application from '../../system/application';
-import ApplicationInstance from '../../system/application-instance';
+import { assign } from 'ember-utils';
+import { moduleFor, AutobootApplicationTestCase, TestResolver } from 'internal-test-helpers';
+import { Application, ApplicationInstance } from 'ember-application';
 import { jQuery } from 'ember-views';
 
-let app;
+moduleFor('Ember.Application instance initializers', class extends AutobootApplicationTestCase {
+  constructor() {
+    jQuery('#qunit-fixture').html(`
+      <div id="one">ONE</div>
+      <div id="two">TWO</div>
+    `);
+    super();
+  }
+
+  createApplication(options, MyApplication) {
+    let myOptions = assign({
+      rootElement: '#one'
+    }, options);
+    super.createApplication(myOptions, MyApplication);
+  }
+
+  createSecondApplication(options, MyApplication) {
+    let myOptions = assign({
+      rootElement: '#two',
+      Resolver: TestResolver
+    }, options);
+
+    let secondApp = this.secondApp = MyApplication.create(myOptions);
+    this.secondResolver = myOptions.Resolver.lastInstance;
+    return secondApp;
+  }
 
-QUnit.module('Ember.Application instance initializers', {
   teardown() {
-    if (app) {
-      run(() => app.destroy());
+    super.teardown();
+
+    if (this.secondApp) {
+      run(this.secondApp, 'destroy');
     }
   }
-});
 
-QUnit.test('initializers require proper \'name\' and \'initialize\' properties', function() {
-  let MyApplication = Application.extend();
+  [`@test initializers require proper 'name' and 'initialize' properties`]() {
+    let MyApplication = Application.extend();
 
-  expectAssertion(() => {
-    run(() => {
-      MyApplication.instanceInitializer({ name: 'initializer' });
+    expectAssertion(() => {
+      run(() => {
+        MyApplication.instanceInitializer({ name: 'initializer' });
+      });
     });
-  });
 
-  expectAssertion(() => {
-    run(() => {
-      MyApplication.instanceInitializer({ initialize() {} });
+    expectAssertion(() => {
+      run(() => {
```

Comment:
As mentioned above, using `this.runTask` instead of `run` is preferred. This allows us to change the runloop abstraction later if we want to. This suggestion applies across the PR.

Here specifically, I don't expect we need a runloop at all. If these assertions are just triggered by `instanceInitializer` the runloop is spurious.